kornia/utils/image.py

```diff
@@ -24,24 +26,50 @@ def image_to_tensor(image: np.ndarray, keepdim: bool = True) -> torch.Tensor:
         raise ValueError(
             "Input size must be a two, three or four dimensional array")
 
-    input_shape = image.shape
     tensor: torch.Tensor = torch.from_numpy(image)
+    # Normalize to channel-last format
+    if len(tensor.shape) == 2:
+        # (H, W) -> (H, W, 1)
+        tensor = tensor.unsqueeze(-1)
+    tensor_shape = tensor.shape
+    tensor = to_bchw(tensor, -1)
+    if keepdim:
+        if len(tensor_shape) == 3:
+            tensor = tensor.reshape(-1, *tensor_shape[:-1])
+        elif len(tensor_shape) == 4:
+            pass
+        else:
+            raise ValueError(f"Unsupported tensor shapes. Got {tensor_shape}.")
+    return tensor
 
-    if len(input_shape) == 2:
-        # (H, W) -> (1, H, W)
-        tensor = tensor.unsqueeze(0)
-    elif len(input_shape) == 3:
-        # (H, W, C) -> (C, H, W)
-        tensor = tensor.permute(2, 0, 1)
-    elif len(input_shape) == 4:
-        # (B, H, W, C) -> (B, C, H, W)
-        tensor = tensor.permute(0, 3, 1, 2)
-        keepdim = True  # no need to unsqueeze
-    else:
-        raise ValueError(
-            "Cannot process image with shape {}".format(input_shape))
 
-    return tensor.unsqueeze(0) if not keepdim else tensor
+def to_bchw(tensor: torch.Tensor, color_channel_num: Optional[int] = None) -> torch.Tensor:
```

Comment:
I don't understand why this function. The name isn't appropriate and the docs + code readability don't help at all. I wouldn't add this to the main functionality. If it's a util for a specific transform make it first there, later if we see that can be useful or generalized we can discuss how adapt core functions to be more generic.

Reply:
Okay. I will make this one private for now. It is essentially doing nothing here but to convert either HW, CHW, HWC or BCHW into BCHW.